tornjak-frontend/src/components/work-load-attestor-modal.tsx

```diff
@@ -1,17 +1,44 @@
 import React from "react";
-import { ModalWrapper, Dropdown, TextArea } from "carbon-components-react";
+import { ModalWrapper, Dropdown, TextArea, DataTableCell } from "carbon-components-react";
 import { connect } from 'react-redux';
 import IsManager from './is_manager';
 import TornjakApi from './tornjak-api-helpers';
 import {
     agentsListUpdateFunc,
     agentworkloadSelectorInfoFunc,
 } from 'redux/actions';
+import { RootState } from 'redux/reducers';
+import { AgentsList, AgentsWorkLoadAttestorInfo } from "./types";
+// import PropTypes from "prop-types";
 
-class WorkLoadAttestor extends React.Component {
-    constructor(props) {
+type WorkLoadAttestorProp = {
+    // dispatches a payload for list of agents with their metadata info as an array of AgentListType and has a return type of void
+    agentsListUpdateFunc: (globalAgentsList: AgentsList[]) => void,
+    // dispatches a payload for the workload selector info for the agents and has a return type of void
+    agentworkloadSelectorInfoFunc: (globalAgentsWorkLoadAttestorInfo: AgentsWorkLoadAttestorInfo[]) => void,
+    spiffeid: string,
+    // the selected server for manager mode 
+    globalServerSelected: string,
+    // list of available agents as array of AgentsListType or can be undefined if no array present
+    globalAgentsList: AgentsList[] | undefined,
```

Comment:
i think there should be a way to enforce that this isnt undefined.

Reply:
Correct, done!